### PR TITLE
Implement array#includes

### PR DESCRIPTION
**Pull request by mateuszmlc**
Includes #97

### CODE REVIEW

File: src/compiler/mod.rs [hunk 1]
```diff
@@ -797,6 +812,36 @@ lazy_static::lazy_static! {
...
+        builtins.push((
+            "includes".to_string(),
+            Some(TypeConstraint::Array),
+            Function::Builtin(Box::new(&|compiler, _, args| {
+                ensure!(args.len() == 2, ArgumentsCountSnafu { found: args.len(), expected: 2usize });
+
+                let old_root_scope = compiler.root_scope;
+                compiler.root_scope = &BUILTINS_SCOPE;
+                let result = array::includes(compiler, &args[0], &args[1])?;
+                compiler.root_scope = old_root_scope;
```
Comment (l4l) [inline src/compiler/mod.rs:839]: Perhaps we should update scope once?

File: src/compiler/array.rs [hunk 2]
```diff
@@ -224,3 +224,158 @@ pub(crate) fn get(compiler: &mut Compiler, arr: &Symbol, index: &Symbol) -> Symb
...
+        index.type_.miden_width(),
+    );
+    // [index]
+    compiler.instructions.extend([
+        Instruction::Push(-1i32 as u32),
+        // [-1, index]
+        Instruction::U32CheckedNeq,
+        // [index != -1]
+        Instruction::MemStore(Some(result.memory_addr)),
+    ]);
```
Comment (l4l) [inline src/compiler/array.rs:378]: According to [mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) it should work similarly to `array.iter().any(|x| x == el)`, no?

Comment (mateuszmlc) [reply src/compiler/array.rs:378]: Yes, except for `NaN`, `indexOf() != -1` and `includes` should work the same. [NaN].includes(NaN) returns true, indexOf will return -1. Rust's `contains` returns false for this, so I don't think it's bad to keep it as `false`.

File: src/compiler/mod.rs (hunk 1 above)
Comment (mateuszmlc) [reply src/compiler/mod.rs:839]: The BUILTINS_SCOPE contains the hidden/system/unsafe builtins and safe builtins. We need to reset it back to the user's old root scope that only has the safe builtins + root functions + collections.

**Review by l4l — APPROVED** [review]

**Pull request merged**